admin/class-extension-manager.php

```diff
@@ -72,20 +76,54 @@ public function get_all() {
 	public function is_activated( $extension_name ) {
 		if ( self::$active_extensions === null ) {
 			// Force re-check on license & dashboard pages.
-			$current_page  = filter_input( INPUT_GET, 'page' );
+			$current_page = $this->get_current_page();
+
+			// Check whether the licenses are valid or whether we need to show notifications.
 			$exclude_cache = ( $current_page === 'wpseo_licenses' || $current_page === 'wpseo_dashboard' );
 
 			// Fetch transient data on any other page.
 			if ( ! $exclude_cache ) {
-				self::$active_extensions = get_transient( self::TRANSIENT_CACHE_KEY );
+				self::$active_extensions = $this->get_cached_extensions();
 			}
 
+			// If the active extensions is still NULL, we need to set it.
 			if ( ! is_array( self::$active_extensions ) ) {
 				self::$active_extensions = (array) apply_filters( 'yoast-active-extensions', array() );
-				set_transient( self::TRANSIENT_CACHE_KEY, self::$active_extensions, DAY_IN_SECONDS );
 			}
+
+			$this->set_cached_extensions( self::$active_extensions );
```

Comment:
Can you put this inside the if. Now it will refresh the transient every request.